Core/include/Acts/Geometry/BlueprintNode.hpp

```diff
@@ -0,0 +1,297 @@
+  /// @param direction The direction of the stack configuration. See
+  ///                  @ref Acts::CylinderVolumeStack for details.
+  /// @param callback An optional callback that receives the node as an argument
+  CylinderContainerBlueprintNode& addCylinderContainer(
+      const std::string& name, BinningValue direction,
+      const std::function<void(CylinderContainerBlueprintNode& cylinder)>&
+          callback = {});
+
+  /// Convenience method for creating a @ref Acts::MaterialDesignatorBlueprintNode.
+  /// @param name The name of the material designator node. Used for debugging
+  ///             the node tree only.
+  /// @param callback An optional callback that receives the node as an argument
+  MaterialDesignatorBlueprintNode& addMaterial(
+      const std::string& name,
+      const std::function<void(MaterialDesignatorBlueprintNode& material)>&
+          callback = {});
+
+  /// Convenience method for creating a @ref Acts::LayerBlueprintNode.
+  /// @param name The name of the layer node.
+  /// @param callback An optional callback that receives the node as an argument
+  LayerBlueprintNode& addLayer(
+      const std::string& name,
+      const std::function<void(LayerBlueprintNode& layer)>& callback = {});
+
+  /// @}
+
+  /// Register a @p child to this node.
+  /// @warning This method throws if adding the child would create a
+  ///          cycle in the blueprint tree!
+  /// @param child The child node to add
+  /// @return A reference this node (not the child!)
+  BlueprintNode& addChild(std::shared_ptr<BlueprintNode> child);
+
+  /// A range-like object that allows range based for loops and index access.
+  /// This type's iterators and accessors return mutable references when
+  /// dereferenced.
+  using MutableChildRange =
+      detail::TransformRange<detail::Dereference,
+                             std::vector<std::shared_ptr<BlueprintNode>>>;
+
+  /// A range-like object that allows range based for loops and index access.
+  /// This type's iterators and accessors return const references when
+  /// dereferenced.
+  using ChildRange =
+      detail::TransformRange<detail::ConstDereference,
+                             const std::vector<std::shared_ptr<BlueprintNode>>>;
+
+  /// Return a @ref MutableChildRange to the children of this node.
+  /// @return A range-like object to the children
+  MutableChildRange children();
+
+  /// Return a @ref ChildRange to the children of this node.
+  /// @return A range-like object to the children
+  ChildRange children() const;
+
+  /// Remove all children from this node
+  void clearChildren();
+
+  /// Return the depth of this node in the blueprint tree. A depth of zero means
+  /// this node does not have a parent.
+  /// @return The depth of this node
+  std::size_t depth() const;
+
+  /// Print the node tree starting from this node to graphviz format
+  /// @param os The stream to print to
+  void graphViz(std::ostream& os) const;
```

Comment:
looks a bit inconsistent vs `addToGraphviz`

Reply:
You mean `graphviz` would be more consistent?